src/python/pants/source/payload_fields.py

```diff
@@ -37,48 +39,51 @@ def source_root(self):
     # this into the target, rather than have it reach out for global singletons.
     return SourceRootConfig.global_instance().get_source_roots().find_by_path(self.rel_path)
 
-  def matches(self, path):
+  def matches(self, path: str) -> bool:
     return self.sources.matches(path) or matches_filespec(path, self.filespec)
 
   @property
-  def filespec(self):
+  def filespec(self) -> Filespec:
     return self.sources.filespec
 
   @property
-  def rel_path(self):
+  def rel_path(self) -> str:
     return self.sources.rel_root
 
   @property
-  def sources(self):
+  def sources(self) -> FilesetWithSpec:
     return self._sources
 
   @memoized_property
-  def source_paths(self):
+  def source_paths(self) -> List[str]:
     return list(self.sources)
 
   @property
   def address(self):
     """Returns the address this sources field refers to (used by some derived classes)"""
     return self._ref_address
 
-  def snapshot(self, scheduler=None):
+  def snapshot(self, scheduler: Optional[SchedulerSession] = None) -> Snapshot:
     """
     Returns a Snapshot containing the sources, relative to the build root.
 
     This API is experimental, and subject to change.
     """
     if isinstance(self._sources, EagerFilesetWithSpec):
-      snapshot = self._sources.snapshot
-      if snapshot is not None:
-        return snapshot
+      return self._sources.snapshot
```

Comment:
`EagerFilesetWithSpec.snapshot` returns `Snapshot`, not `Optional[Snapshot]`, so it was unnecessary to check if it is `None`.